File: Java/SpaceVIL/spacevil/src/main/java/com/spvessel/spacevil/Frame.java
package com.spvessel.spacevil;

import com.spvessel.spacevil.Common.DefaultsService;

/**
 * Frame is the basic container. It groups items based on items alignment,
 * margins, paddings, sizes and size policies.
 * <p>
 * Frame cannot receive any events, so Frame is always in the
 * com.spvessel.spacevil.Flags.ItemStateType.BASE state.
 */
public class Frame extends Prototype {
    private static int count = 0;

    /**
     * Default Frame constructor. Frame cannot get focus.
     */
    public Frame() {
        setItemName("Frame_" + count);
        count++;
        setStyle(DefaultsService.getDefaultStyle(Frame.class));
        isFocusable = false;
    }

    @Override
    protected boolean getHoverVerification(float xpos, float ypos) {
        return false;
    }
}

File: SpaceVIL/Items/Frame.cs
using SpaceVIL.Common;

namespace SpaceVIL
{
    /// <summary>
    /// Frame is the basic container. 
    /// It groups items based on items alignment, margins, paddings, sizes and size policies.
    /// <para/> Frame cannot receive any events, so Frame is always in the SpaceVIL.Core.ItemStateType.Base state.
    /// </summary>
    public class Frame : Prototype
    {
        static int count = 0;

        /// <summary>
        /// Default Frame constructor. Frame cannot get focus.
        /// </summary>
        public Frame()
        {
            SetItemName("Frame_" + count);
            count++;

            SetStyle(DefaultsService.GetDefaultStyle(typeof(SpaceVIL.Frame)));
            IsFocusable = false;
        }

        protected internal override bool GetHoverVerification(float xpos, float ypos)
        {
            return false;
        }
    }
}
